///////////////////////////////////////////////////
//         Define Protobuf Syntax Version        //         
///////////////////////////////////////////////////
syntax = "proto3";

///////////////////////////////////////////////////
//              Define Package Name              //         
///////////////////////////////////////////////////
package spi;

///////////////////////////////////////////////////
//               Define RPC Service              //         
///////////////////////////////////////////////////
service SPI {
    // Configure DAC Channel 0
    rpc ConfigureDACChannel0(ConfigureDACChannel0Packet) returns (ConfigureDACChannel0Response) {}
    // Configure DAC Channel 1
    rpc ConfigureDACChannel1(ConfigureDACChannel1Packet) returns (ConfigureDACChannel1Response) {}
    // Configure DAC Channel 2
    rpc ConfigureDACChannel2(ConfigureDACChannel2Packet) returns (ConfigureDACChannel2Response) {}
}

///////////////////////////////////////////////////
//              Define Message Type              //         
///////////////////////////////////////////////////
// Configure DAC Channel 0 Message
message ConfigureDACChannel0Packet {}

// Configure DAC Channel 0 Response
message ConfigureDACChannel0Response {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Configure DAC Channel 1 Message
message ConfigureDACChannel1Packet {}

// Configure DAC Channel 1 Response
message ConfigureDACChannel1Response {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Configure DAC Channel 2 Message
message ConfigureDACChannel2Packet {}

// Configure DAC Channel 2 Response
message ConfigureDACChannel2Response {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}
